src/cdk-experimental/table-scroll-container/table-scroll-container.ts

```diff
@@ -0,0 +1,136 @@
+      @Optional() private readonly _directionality: Directionality,
+      elementRef: ElementRef<HTMLElement>,
+      @Inject(DOCUMENT) document: any) {
+    this._document = document;
+    this._element = elementRef.nativeElement;
+    this._uniqueClassName = `cdk-table-scroll-container-${++nextId}`;
+    this._element.classList.add(this.CLASS_NAME);
+    this._element.classList.add(this._uniqueClassName);
+  }
+
+  ngOnDestroy(): void {
+    // TODO: Use remove() once we're off IE11.
+    if (this._styleElement?.parentNode) {
+      this._styleElement.parentNode.removeChild(this._styleElement);
+      this._styleElement = undefined;
+    }
+  }
+
+  stickyColsUpdated(sizes: (number|null|undefined)[]): void {
+    this._startSizes = sizes;
+    this._updateHorizontal();
+  }
+
+  stickyEndColsUpdated(sizes: (number|null|undefined)[]): void {
+    this._endSizes = sizes;
+    this._updateHorizontal();
+  }
+
+  stickyHeaderRowsUpdated(sizes: (number|null|undefined)[]): void {
+    this._headerSizes = sizes;
+    this._updateVertical();
+  }
+
+  stickyFooterRowsUpdated(sizes: (number|null|undefined)[]): void {
+    this._footerSizes = sizes;
+    this._updateVertical();
+  }
+
+  private _updateHorizontal(): void {
+    const startMargin = computeMargin(this._startSizes);
+    const endMargin = computeMargin(this._endSizes);
+
+    if (startMargin === 0 && endMargin === 0) {
+      this._applyCss('horizontal', '');
+      return;
+    }
+
+    const direction = this._directionality ? this._directionality.value : 'ltr';
+    const leftMargin = direction === 'rtl' ? endMargin : startMargin;
+    const rightMargin = direction === 'rtl' ? startMargin : endMargin;
+
+    this._applyCss('horizontal', `0 ${rightMargin}px 0 ${leftMargin}px`);
+  }
+
+  private _updateVertical(): void {
+    const topMargin = computeMargin(this._headerSizes);
+    const bottomMargin = computeMargin(this._footerSizes);
+
+    if (topMargin === 0 && bottomMargin === 0) {
+      this._applyCss('vertical', '');
+      return;
+    }
+
+    this._applyCss('vertical', `${topMargin}px 0 ${bottomMargin}px`);
+  }
+
+  private _getStyleSheet(): CSSStyleSheet {
+    if (!this._styleElement) {
+      this._styleElement = this._document.createElement('style');
+      this._styleElement.appendChild(this._document.createTextNode(''));
+      this._document.head.appendChild(this._styleElement);
+    }
+
+    return this._styleElement.sheet as CSSStyleSheet;
+  }
+
+  private _applyCss(orientation: Orientation, value: string) {
+    let index = this._ruleIndexes.get(orientation);
+    if (index === undefined) {
+      if (!value) {
+        return;
+      }
+
+      index = this._indexSequence++;
+      this._ruleIndexes.set(orientation, index);
+    } else {
+      this._getStyleSheet().deleteRule(index);
+    }
+
+    const selector = `.${this._uniqueClassName}::-webkit-scrollbar-track:${orientation}`;
+    this._getStyleSheet().insertRule(`${selector} {margin: ${value}}`, index!);
```

Comment:
Doing this through a global stylesheet won't work if the table is inside a component with `ViewEncapsulation.ShadowDom`.

Reply:
That's true... Something I've been wanting for Angular generally is an Injectable that gives you either the document or the current shadow root.

I suppose as another alternative, I could insert the stylesheet as a sibling node to this directive.

Reply:
There's a utility called `_getShadowRoot` that will give you the closest shadow root node. The gotcha is that you don't want to call it too early or the element might not be in the shadow DOM yet.